### PR TITLE
Feature/upgrade in future maintenance window

### PR DESCRIPTION
**Pull request by 0xE282B0**
Description
Schedule upgrades in future maintenance window

Changes proposed in this pull request:

Adds --scheduleAfter parameter to cli
KEB can schedule upgrades to maintenance windows later than the next one
Related issue(s)

[GHE/backlog#2480](https://github.tools.sap/kyma/backlog/issues/2480)

### CODE REVIEW

**Comment by piotrmiskiewicz** [comment]
Hi @0xE282B0 , for me it looks ok, but I'd like someone from SRE team give the green light (then I put an approval)

File: components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go [hunk 1]
```diff
@@ -74,6 +74,22 @@ func (h *kymaHandler) createOrchestration(w http.ResponseWriter, r *http.Request
...
+	}
+
+	// validate `schedule` field
+	err = h.ValidateScheduleParameter(&params)
+	if err != nil {
+		h.log.Errorf("found deprecated value: %v", err)
+		httputil.WriteErrorResponse(w, http.StatusBadRequest, errors.Wrapf(err, "found deprecated value"))
+		return
+	}
+
```
Comment (ebensom) [inline components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go:92]: I would extract and move these to the handlers.go as this conversion/validation should also be used by cluster upgrade handler.

File: components/kyma-environment-broker/internal/orchestration/handlers/handlers.go [hunk 2]
```diff
@@ -51,10 +51,5 @@ func defaultOrchestrationStrategy(spec *orchestration.StrategySpec) {
 		spec.Type = orchestration.ParallelStrategy
 	}
 
-	switch spec.Schedule {
-	case orchestration.MaintenanceWindow:
-	case orchestration.Immediate:
-	default:
-		spec.Schedule = orchestration.Immediate
-	}
+	spec.Schedule = string(orchestration.Immediate)
```
Comment (ebensom) [inline components/kyma-environment-broker/internal/orchestration/handlers/handlers.go:54]: I would remove the `defaultOrchestrationStrategy` function, and force client to provide the immediate/now or date with the validation functions you implemented in kyma handler (which would be moved here)

File: components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go (hunk 1 above)
Comment (0xE282B0) [reply components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go]: Moved the validation methods to `handlers.go` and added validation to `cluster_handler.go`.

File: components/kyma-environment-broker/common/orchestration/strategies/parallel_test.go [hunk 3]
```diff
@@ -74,7 +74,7 @@ func TestNewParallelOrchestrationStrategy_MaintenanceWindow(t *testing.T) {
 	}
 
 	// when
-	id, err := s.Execute(ops, orchestration.StrategySpec{Schedule: orchestration.MaintenanceWindow, Parallel: orchestration.ParallelStrategySpec{Workers: 2}})
+	id, err := s.Execute(ops, orchestration.StrategySpec{Schedule: "imideate", MaintenanceWindow: true, Parallel: orchestration.ParallelStrategySpec{Workers: 2}})
```
Comment (ebensom) [inline components/kyma-environment-broker/common/orchestration/strategies/parallel_test.go:77]: Fun :D
```suggestion
	id, err := s.Execute(ops, orchestration.StrategySpec{Schedule: "immediate", MaintenanceWindow: true, Parallel: orchestration.ParallelStrategySpec{Workers: 2}})
```

**Review by ebensom — APPROVED** [review]

**Pull request merged**